### PR TITLE
[WIP] update K8s deps to 1.16.1

### LINKED ISSUE #937: Update kubernetes libraries to v1.16.x
Now that the `1.16` is the latest stable minor release of kubernetes, we should be vendoring them to kube-state-metrics.

### PR DESCRIPTION
**Pull request by brennerm**
**What this PR does / why we need it**:
This PR updates the K8s dependencies to 1.16.1.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #937

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,34 +1,26 @@
 module k8s.io/kube-state-metrics
 
 require (
+	github.com/Azure/go-autorest v11.1.2+incompatible // indirect
```
Comment (tariq1890) [inline go.mod]: remove this line. We don't need it.

File: go.mod [hunk 2]
```diff
@@ -1,34 +1,26 @@
...
-	k8s.io/api v0.0.0-20190620084959-7cf5895f2711
-	k8s.io/apimachinery v0.0.0-20190612205821-1799e75a0719
-	k8s.io/autoscaler v0.0.0-20190607113959-1b4f1855cb8e
-	k8s.io/client-go v0.0.0-20190620085101-78d2af792bab
+	golang.org/x/tools v0.0.0-20190312170243-e65039ee4138
+	k8s.io/api v0.0.0-20191003000013-35e20aa79eb8
+	k8s.io/apimachinery v0.0.0-20190913080033-27d36303b655
+	k8s.io/autoscaler v0.0.0-20190923084523-b8ef369fd110
+	k8s.io/client-go v0.0.0-20191003000419-f68efa97b39e
 	k8s.io/klog v0.4.0
```
Comment (tariq1890) [inline go.mod]: Let's update klog to v1.0.0

**Comment by tariq1890** [comment]
@brennerm Once you have addressed the comments, please run `go mod tidy`, `go mod vendor` and then check-in the vendored files.

**Comment by brennerm** [comment]
Meaning we'll have to wait for it to become compatible?

I also tried updating to 1.16.0, but it doesn't work either.

**Comment by tariq1890** [comment]
Does it work when you don't update cluster-autoscaler?

**Comment by brennerm** [comment]
Nope, still the same error.

**Comment by tariq1890** [comment]
@brennerm We'll have to wait for the issue to get resolved - https://github.com/kubernetes/autoscaler/issues/2447

Once that's done, you should be able to move forward.

**Comment by ymmt2005** [comment]
@brennerm @tariq1890 
https://github.com/kubernetes/autoscaler/issues/2447 has been resolved.  Could you move forward?

**Comment by brennerm** [comment]
@ymmt2005 From my understanding the the PR #949 takes care of updating the deps already.

@wozniakjan Please correct me if I'm wrong. If so I'll happily move forward with this PR.

**Comment by wozniakjan** [comment]
@brennerm correct, I took the liberty of adding the updated deps to the PR directly for testing purposes, but in case we would like to keep as two distinct and separate PRs, happy to rebase on top of this as soon this passes tests and is ready to merge.

But yeah, should be sufficient to merge #949 if that is ok.

**Comment by brennerm** [comment]
@wozniakjan Totally fine for me. :+1:

**Pull request closed without merging**